staging/src/k8s.io/api/admissionregistration/v1alpha1/types.go

```diff
@@ -0,0 +1,375 @@
+	// The message must not contain line breaks.
+	// If unset, the message is "failed Expression: {Expression}".
+	// +optional
+	Message string `json:"message" protobuf:"bytes,3,opt,name=message"`
+	// If the MessageExpression evaluates to an error, the Message field is used to provide the message.
+	// The MessageExpression retured when failed the validation of specified.
+	// If MessageExpression is specified, Message will be ignored
+	// If the Expression contains line breaks. Either Message or MessageExpression is required.
+	// The MessageExpression must not contain line breaks.
+	// +optional
+	MessageExpression string `json:"expressMessage" protobuf:"bytes,4,opt,name=expressMessage"`
+	// Reason returned when failed the validation.
+	// +optional
+	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// Code returned when failed the validation.
+	Code string `json:"code" protobuf:"bytes,6,opt,name=code"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.26
+
+// PolicyBinding binds the ValidatingAdmissionPolicy with paramerized resources.
+// PolicyBinding and parameter CRDs together define how cluster administrators configure policies for clusters.
+type PolicyBinding struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Specification of the desired behavior of the AdmissionPolicy.
+	// +optional
+	Spec PolicyBindingSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:prerelease-lifecycle-gen:introduced=1.26
+
+// PolicyBindingList is a list of PolicyBinding.
+type PolicyBindingList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// List of PolicyBinding.
+	Items []PolicyBinding `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// PolicyBindingSpec is the specification of the PolicyBinding.
+type PolicyBindingSpec struct {
+	// Policy references a ValidatingAdmissionPolicy which the policyBinding binds to.
+	// Required.
+	Policy string `json:"policy,omitempty" protobuf:"bytes,1,rep,name=policy"`
+
+	//Param specifies the parameter resource used to configure the admission control policy.
+	// It should point to a Customer Resource which is created out of the CRD specified in ParamSource of ValidatingAdmissionPolicy it binded.
+	// +optional
+	Param string `json:"param,omitempty" protobuf:"bytes,2,rep,name=param"`
```

Comment:
Do we want `param` or `params`?  The referenced custom resource is intended to contain parameters, so plural makes sense to me, but I'm curious what others think.

Reply:
I guess param is ok since we take string here. Params might cause confusion of taking []string

Reply:
`ParamsResourceName`?